# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: tools/profiler/core/ProfilerBacktrace.h
# Commit: b6d73f5042a7
# Full Hash: b6d73f5042a7f3703b5d16e222e9f71153a43164
# Author: Gerald Squelart <gsquelart@mozilla.com>
# Date: 2019-09-17 15:54:53
# Regressor Bug: 1576551
# File Overlap Count: 1
# Description:
#   Bug 1576551 - Store BlocksRingBuffer outside of ProfileBuffer - r=gregtatum
#   
#   The main `BlocksRingBuffer`s will be stored in `CorePS` (outside of
#   `ProfileBuffer`s), as we need to be able to safely access the underlying buffers
#   when profilers are not enabled.
# ==============================================================================

diff -r 366030bd2d84 -r b6d73f5042a7 tools/profiler/core/ProfilerBacktrace.h
--- a/tools/profiler/core/ProfilerBacktrace.h	Tue Sep 17 01:50:19 2019 +0000
+++ b/tools/profiler/core/ProfilerBacktrace.h	Tue Sep 17 02:12:42 2019 +0000
@@ -16,14 +16,16 @@
 class UniqueStacks;
 
 namespace mozilla {
+class BlocksRingBuffer;
 class TimeStamp;
-}
+}  // namespace mozilla
 
 // ProfilerBacktrace encapsulates a synchronous sample.
 class ProfilerBacktrace {
  public:
   ProfilerBacktrace(const char* aName, int aThreadId,
-                    mozilla::UniquePtr<ProfileBuffer> aBuffer);
+                    UniquePtr<mozilla::BlocksRingBuffer> aBlocksRingBuffer,
+                    mozilla::UniquePtr<ProfileBuffer> aProfileBuffer);
   ~ProfilerBacktrace();
 
   // ProfilerBacktraces' stacks are deduplicated in the context of the
@@ -42,7 +44,10 @@
 
   mozilla::UniqueFreePtr<char> mName;
   int mThreadId;
-  mozilla::UniquePtr<ProfileBuffer> mBuffer;
+  // `BlocksRingBuffer` in which `mProfileBuffer` stores its data; must be
+  // located before `mProfileBuffer` so that it's destroyed after.
+  UniquePtr<mozilla::BlocksRingBuffer> mBlocksRingBuffer;
+  mozilla::UniquePtr<ProfileBuffer> mProfileBuffer;
 };
 
 #endif  // __PROFILER_BACKTRACE_H